web-marketplace/src/components/organisms/ChooseCreditsForm/ChooseCreditsForm.tsx

```diff
@@ -0,0 +1,209 @@
+import { RadioGroup } from 'web-components/src/components/inputs/new/RadioGroup/RadioGroup';
+import { Loading } from 'web-components/src/components/loading';
+import Title from 'web-components/src/components/typography/new/Title';
+
+import { PaymentOptions } from './ChooseCreditsForm.PaymentOptions';
+import {
+  ChooseCreditsFormSchema,
+  FormFields,
+} from './ChooseCreditsForm.schema';
+import { PAYMENT_OPTIONS, PaymentOptionsType } from './ChooseCreditsForm.types';
+
+export function ChooseCreditsForm({
+  creditVintages,
+  cryptoOptions,
+}: {
+  creditVintages: { date: string; credits: string }[];
+  cryptoOptions: { label: string; description?: string; linkTo: string }[];
+}) {
+  const [paymentOption, setPaymentOption] = useState<PaymentOptionsType>(
+    PAYMENT_OPTIONS.CARD,
+  );
+  const [advanceSettingsOpen, setAdvanceSettingsOpen] = useState(false);
+  const [creditVintageOptions, setCreditVintageOptions] = useState<
+    string[] | []
+  >([]);
+
+  const toggleAdvancedSettings = (e: MouseEvent<HTMLElement>) => {
+    e.preventDefault();
+    setAdvanceSettingsOpen(prev => !prev);
+  };
+
+  const form = useZodForm({
+    schema: ChooseCreditsFormSchema,
+    defaultValues: {
+      amountCurrency: 1,
+      amountCredits: 1,
+      cryptoPurchaseOption: cryptoOptions?.[0]?.label,
+    },
+    mode: 'onBlur',
+  });
+
+  const cryptoPurchaseOption = useWatch({
+    control: form.control,
+    name: 'cryptoPurchaseOption',
+  });
+
+  const handleOnSubmit: SubmitHandler<FormFields> = data => {
+    // TO-DO
+  };
+
+  const handleCryptoPurchaseOptions = (e: ChangeEvent<HTMLInputElement>) => {
+    form.setValue('cryptoPurchaseOption', e.currentTarget.value);
+  };
+
+  const handleCreditVintageOptions = (e: ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+    const checked = e.target.checked;
+    const updatedItems = checked
+      ? [...creditVintageOptions, value]
+      : creditVintageOptions.filter(item => item !== value);
+    form.setValue('creditVintageOptions', updatedItems);
+    setCreditVintageOptions(updatedItems);
+  };
+
+  const handlePaymentOptions = (option: string) => {
+    setPaymentOption(option as PaymentOptionsType);
+  };
+
+  return (
+    <Suspense fallback={<Loading />}>
+      <div
+        data-testid="buy-credits-form"
+        className="border-grey-300 border border-solid p-15 sm:p-20 rounded-[5px]"
+      >
+        <Form form={form} onSubmit={handleOnSubmit}>
+          {/* payment options */}
+          <PaymentOptions setPaymentOption={handlePaymentOptions} />
+
+          {/* credits amount */}
+          <div
+            className={`grid min-h-min ${
+              paymentOption === PAYMENT_OPTIONS.CRYPTO
+                ? 'grid-rows-3'
+                : 'grid-rows-2'
+            }`}
+          >
+            <CreditsAmount
+              creditsAvailable={1500} // TO-DO update with max credits available {creditsAvailable}
```

Comment:
I believe this could be done in this PR, see my previous comment about adding more props to this component
the number of credits available should change when the selected currency changes

Reply:
I've updated this based on your feedback. Please let me know if these changes is what you had in mind.

Reply:
looks good